### DISCUSSION TITLE
Migrating to lldap from Authelia file provider

### DISCUSSION THREAD

**Discussion by mpfl** [lldap/lldap#272] [Q&A] [answered]
I'm a happy user of Authelia, but currently use the ["file" provider for first-factor](https://www.authelia.com/configuration/first-factor/file/) (password). I'd much rather use lldap for my ~5 users but would like to avoid asking each one to reset their password.

Is there an elegant way to import the data from my Authelia file ([which looks like this example](https://www.authelia.com/reference/guides/passwords/#user--password-file)) into lldap?

If there's no elegant way, is it possible to inject the argon2 hashes into the lldap database after I manually create the users?

**Comment by zicklag** [comment]
I don't have a script or anything, but I believe lldap has a GraphQL API that could be used to script out the conversion.

**Comment by nitnelave** [comment, marked as answer]
You can have a look at the migration tool (in Rust) in the repo. Currently
it only supports importing from OpenLDAP, but you could add support for
authelia if you wanted.

Note on importing password hashes: that's not possible because LLDAP
doesn't just use password hashing, it actually uses OPAQUE which is a zero
knowledge cryptographic proof that you have the right password without
actually sending it to the server.
